### DISCUSSION TITLE
Products with Quantity Discounts that are also on sale

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#5775] [Ideas]
It's unclear to me as to whether the subject combination is supported, but ...

zc156b, demo products.  I modified product id=175 (Quantity Discounts by 1) to have a 50%-off sale.  The admin's Products Price Manager displays:

![image](https://user-images.githubusercontent.com/2685585/64727258-6ecb7d00-d4a6-11e9-90d0-2620801754fc.png)

When displayed on the storefront, however, the 50%-off sale price is applied _only to_ the first of the quantity discounts:
![image](https://user-images.githubusercontent.com/2685585/64727386-bbaf5380-d4a6-11e9-8d8e-bc472659fec8.png)

**Comment by mc12345678** [comment]
I found that when the setting for Discount Priced From is changed to Special that the quantity discounts then discount against the calculated sale price.

So in the above example that if purchasing a quantity of two then first a 50% discount would be taken off of the base price of 60 giving 30, then a 3% discount would be taken off of that to identify that each would be 29.1 for a total cost of 58.2 for the two items.

I haven't tested to see what happens if the item has both a sale and a special (ie. Which price would be used) and whether one or the other results in a higher price instead of progressively lower. Seems a sale applies to a category where a special applies to a product... Starts to get a little crazy when considering much beyond that point... :) Need to really sit down and figure out what might be expected to happen at that/those situations...

**Comment by mc12345678** [comment]
Curious, is this still considered an issue if the desired result is achieved when the quantity discount is priced from special instead of product? Or does the text of that dropdown need to be modified to reflect say `Special/Sale`?

**Comment by lat9** [comment]
Yes, if the discount type is a fixed amount.

**Comment by mc12345678** [comment]
> Yes, if the discount type is a fixed amount.

When referencing a "fixed amount" is that when Discount Type is: "Actual Price" or "Amount Off"?

I understand these to be direct values and their impact is not adjusted by the presence/existence of a sale/special.  With "Amount Off" I see whether the option of Discount Priced from: `Product` or `Special` that the value obtained is reduced by the entered `Discount Value`.  For `Product` the sale/special impacts only the first quantity after that the purchase price is based off of the original product price.  For `Special` I see that the resulting price is reduced from whatever the sale/special price is.

With a Discount Type of: `Actual Price`, the sale/special basically has no impact on anything matching the minimum quantity criteria and the product is priced at the entered `Discount Value`.

Unfortunately for values not entered as a percentage and also associated with the special price, then when the quantity discount is applied to a product that is on sale/special, the result can be more drastic of a reduction or none at all.

Not sure what issue is being seen/needing addressing.  My recommendation for setting up a quantity discount with the potential for sales/specials to be applied to the product would be to determine the percentage that provides the desired price, mark the quantity discount as applicable to the special price and that way when a special is applied a relational discount will be provided.  Alternatively it may be "neat" to offer some sort of on screen "calculator" where one could select actual price, enter that price then when selecting percentage that it convert that value into the associated percentage and allow saving that value.  This way changes could be made with some relational reference and not having to depend on an actual calculator and transcription errors.  Just a thought.

**Comment by lat9** [comment]
It's amount off.  I'm thinking that the application of a sale (or special) should consider its affect on any amount-off/fixed-price quantity discounts.

There are stores that have many, many products and a store-admin can't be expected to remember all those values!

**Comment by mc12345678** [comment]
Fyi, I recently came across something slightly related to this, though not 100% sure directly so but may end up with an equivalent fix.

Product was priced by attribute, each attribute had a different price and the product supported a mix of attributes to support meeting the quantity. (It doesn't seem though that the application of min/max mixing is specifically a cause of the following). A quantity discount was applied as a percentage off when x product are purchased.

If any attribute other than the base/lowest priced attribute was selected and any quantity less than the breakpoint was selected, then the price of the product was equal to the base price and not the price associated with the chosen attribute.

If an entry were added to the quantity discount table where a quantity of 1 (for product sold in whole units) resulted in a 0 percent change in the price, then the applicable price of the attribute would carry through.

The major problem with this at least in display to the customer is that with the added 0 percent reduction, then the table for the discounts shows: 1:base price, 1-x: base price, x+: reduced price. Where the 1-base price seems superfluous.

Not sure if there is a need for it to operate the way it did/does without the 0% reduction entry or if the store should automatically consider the priced-by-attribute price as selected automatically without this additional entry. If that additional entry should be considered necessary to be included, then it seems that the display shouldn't be so separated to target value 1 and then the range of 1-x product... or it should be something that is more easily controllable...
